tests/net/deckard/matches.rs

```diff
@@ -83,16 +85,47 @@ where
             sections.authority.clone(),
             msg.authority().unwrap(),
             msg.header_counts().nscount(),
+            matches.ttl,
             verbose,
         )
     {
         if verbose {
-            todo!();
+            println!("match_msg: authority section does not match");
+        }
+        return false;
+    }
+    if matches.ad && !msg.header().ad() {
+        if verbose {
+            println!("match_msg: AD not in message",);
+        }
+        return false;
+    }
+    if matches.cd && !msg.header().cd() {
+        if verbose {
+            println!("match_msg: CD not in message",);
         }
         return false;
     }
     if matches.fl_do {
-        todo!();
+        if let Some(opt) = msg.opt() {
+            if !opt.dnssec_ok() {
+                if verbose {
```

Comment:
Maybe it's worth writing a tiny macro called `verbose_println!()` ?

Reply:
Or use logging for it throughout?

Reply:
PR #274 introduces logging via tokio tracing and indeed does some of that for Deckard/Stelline, so we could maybe revisit this after looking at that?

Reply:
The function is called in two different ways. One is to find a suitable entry to respond, that one should not log failures. The other is when a reply is expected by the client and failures need to be logged.